README.md

```diff
@@ -179,3 +179,22 @@ will not be emitted by `onEvent`. For more information, see the docs.
  * `RemoteRoom` - A representation of a third-party room.
  * `MatrixUser` - A representation of a matrix user.
  * `RemoteUser` - A representation of a third-party user.
+
+
+## Signaling Bridge Errors
+This section applies when you are using `Bridge` and want to notify your users
+about problems while processing their events.
+
+One thing the bridge requires you to do is fulfilling or rejecting the
+`request` promise which is handed to you as argument of the
+`controller.onEvent` callback. When rejecting the promise the `Error` you
+reject with does matter:
+
+- On most `Error`s the bridge will simply notify the homeserver that the event
+  could not be handled right now. The homeserver will then try to resend the
+  message again with exponential backoff.
```

Comment:
This sounds strange to me. Would an Error cause the bridge to emit a non-200 status code to the homeserver in order to make the homeserver resend it. It sounds dangerous to me to start feeding bridges the same events, as I believe there may be bridges at present which return `Error` for minor things, while the event may have been processed.

I must have missed this in my previous review, but I would urge that Errors are handled the same way as successes and just use your custom Error types to influence bridge logic.

Reply:
Reread some docs and code. I got that part wrong >.< The AS (matrix-appservice-node) will always respond with a 200 as soon as the callback finishes (regardless of what its returned promise is). The event is then irrevocable the problem of the bridge.

That makes me want to reconsider some things about the MSC. Even with that misunderstanding, I think the code is still ok in this form for now.

Reply:
>Reread some docs and code. I got that part wrong >.< The AS (matrix-appservice-node) will always respond with a 200 as soon as the callback finishes (regardless of what its returned promise is). The event is then irrevocable the problem of the bridge.

Ah good. I was worried you may have changed this and I missed it. Anyway that's cool. I guess you can update this README statement and then the code should be good :)